jupyterlab_pullrequests/managers/manager.py

```diff
@@ -203,7 +207,7 @@ async def _call_provider(
                             break
 
                 new_ = json.loads(result)
-                if next_url is not None:
+                if has_pagination and next_url is not None:
```

Comment:
`has_pagination` should not be added - but if you have a better API idea it would be great.

As the `next_url` contains already all required parameters, `has_pagination` is passed as `False` when calling the next page (see line 217). But if there are more than 2 pages, requiring `has_pagination` to be `True` will block calling the page 3, 4,... .

Unfortunately bringing the pagination in the frontend will be tricky to pull all the discussions.

Reply:
yeah, unfortuntately i don' think we can just always assume it's okay to pull down 1000 PRs indiscriminantly...